src/main/java/com/wire/helium/API.java

```diff
@@ -374,53 +510,66 @@ public ArrayList<Integer> getAvailablePrekeys(String clientId) {
+     */
     @Override
-    public Collection<User> getUsers(Collection<UUID> ids) {
-        return usersPath.
-                queryParam("ids", ids.toArray()).
-                request(MediaType.APPLICATION_JSON).
-                header(HttpHeaders.AUTHORIZATION, bearer(token)).
-                get(new GenericType<>() {
-                });
+    public Collection<User> getUsers(Collection<QualifiedId> ids) {
+        throw new UnsupportedOperationException("Bot specific feature, use a more specific API implementation");
     }
 
-    public User getUser(UUID userId) throws HttpException {
-        Response response = usersPath.
-                path(userId.toString()).
-                request(MediaType.APPLICATION_JSON).
-                header(HttpHeaders.AUTHORIZATION, bearer(token)).
-                get();
+    /**
+     * Unused in base api, only needed for bot specific purposes which already extend this API.
+     * @param userId user id
+     */
+    @Override
+    public User getUser(QualifiedId userId) throws HttpException {
+        throw new UnsupportedOperationException("Bot specific feature, use a more specific API implementation");
+    }
 
-        if (response.getStatus() != 200) {
-            throw new HttpException(response.readEntity(String.class), response.getStatus());
-        }
+    public NotificationList retrieveNotifications(String client, UUID since, int size) throws HttpException {
+        WebTarget webTarget = notificationsPath
+                .queryParam("client", client)
+                .queryParam("size", size);
 
-        return response.readEntity(User.class);
-    }
+        if (since != null) {
+            webTarget = webTarget
+                    .queryParam("since", since.toString());
+        }
 
-    public UUID getUserId(String handle) throws HttpException {
-        Response response = usersPath.
-                path("handles").
-                path(handle).
-                request(MediaType.APPLICATION_JSON).
-                header(HttpHeaders.AUTHORIZATION, bearer(token)).
-                get();
+        Response response = webTarget
+                .request(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON)
+                .header(HttpHeaders.AUTHORIZATION, bearer(token))
+                .get();
 
-        if (response.getStatus() != 200) {
-            throw new HttpException(response.readEntity(String.class), response.getStatus());
+        int status = response.getStatus();
+
+        if (status == 200) {
+            return response.readEntity(NotificationList.class);
+        } else if (status == 404) {
+            final NotificationList emptyNotifications = new NotificationList();
+            emptyNotifications.hasMore = false;
+            emptyNotifications.notifications = new ArrayList<>();
+            return emptyNotifications;
+        } else if (status == 401) {   //todo nginx returns text/html for 401. Cannot deserialize as json
```

Comment:
Is this still a TODO?

Reply:
Nope, I checked again and that is indeed still true, so removing the TODO part and left the explanation